x-pack/platform/plugins/private/upgrade_assistant/public/application/components/es_deprecations/deprecation_types/indices/flyout/steps/details/unfreeze_details_step.tsx

```diff
@@ -151,33 +152,44 @@ export const UnfreezeDetailsFlyoutStep: React.FunctionComponent<{
                   </EuiText>
                 ),
               },
+              /* We cannot unfreeze backing indices (that would break the related data_stream) */
+              ...(!isInDataStream
+                ? [
+                    {
+                      title: i18n.translate(
+                        'xpack.upgradeAssistant.esDeprecations.indices.indexFlyout.detailsStep.unfreeze.option2.title',
+                        {
+                          defaultMessage: 'Option 2: Reindex data',
+                        }
+                      ),
+                      description: (
+                        <EuiText size="m">
+                          <FormattedMessage
+                            id="xpack.upgradeAssistant.esDeprecations.indices.indexFlyout.detailsStep.unfreeze.option2.description"
+                            defaultMessage="Alternatively, you can reindex the data into a new, compatible index. All existing documents will be copied over to a new index, and the old index will be removed. Depending on the size of the index and the available resources, the reindexing operation can take some time. Your data will be in read-only mode until the reindexing has completed."
+                          />
+                          {isClosedIndex && (
```

Comment:
Nice! This addresses my concern of showing this warning when the reindex option is not present.